oc-game-swift/Players/Player.swift

```diff
@@ -36,35 +42,166 @@ class Player {
             // Read the answer
             if let name = readLine(){
-                // If the name is not empty, add the hero to team
-                if !name.isEmpty {
+                // Check if the name is already taken
+                if Game.heroesNames.contains(name) {
+                    print("This name is already taken")
+                } else if name.isEmpty{
+                    // Check if the name is empty
+                    print("Srsly? Everytime? Choose a name...")
+                } else { // If not
                     // Switch for the type of hero to add
                     switch choice {
                     case .fighter:
                         // Add a Fighter
-                        player.heroTeam.append(Fighter(name: name, type: .fighter))
+                        player.heroTeam.append(Fighter(name: name))
                     case .healer:
                         // Add a Healer
-                        player.heroTeam.append(Healer(name: name, type: .healer))
+                        player.heroTeam.append(Healer(name: name))
                     case .dwarf:
                         // Add a Dwarf
-                        player.heroTeam.append(Dwarf(name: name, type: .dwarf))
+                        player.heroTeam.append(Dwarf(name: name))
                     case .colossus:
                         // Add a Colossus
-                        player.heroTeam.append(Colossus(name: name, type: .colossus))
+                        player.heroTeam.append(Colossus(name: name))
                     }
                     // Print that all is good
                     print("Your \(choice) \(name) has been added to your team")
+                    // Add the name to the array of heroes names
+                    Game.heroesNames.append(name)
                     // Turn the check on true
                     heroWasCreated = true
-                    
-                } else {
-                    // If the name is empty
-                    print("Srsly? Everytime? Choose a name...")
-                    
                 }
             }
         }
     }
     
+    /// Choosing a hero
+    /// - Player have to choose a hero to fight with this round
+    /// - return: The selected hero
+    func chooseHeroFromYourTeam() -> Hero? {
+        // Hero is choose ?
+        var heroChosen = false
+        // While not
+        while heroChosen == false {
+            // Print the message
+            print("-----------------------------------"
+                + "\n\(self.name), it's your turn. Choose a hero to fight with"
+                + "\n-----------------------------------"
+                + "\n1. \(self.heroTeam[0].description())"
+                + "\n2. \(self.heroTeam[1].description())"
+                + "\n3. \(self.heroTeam[2].description())")
+            // Choose a hero, and check if its ok
+            if let hero = chooseHero(self) {
+                heroChosen = true
```

Comment:
you don't need to use a variable `heroChosen`, have you tried to do something like this : 
```swift
while true {
  //...
  if let hero = chooseHero(self) {
    return hero
  }
}
return

Reply:
Done, yes its works :)